src/main/java/apoc/meta/Meta.java

```diff
@@ -443,12 +446,14 @@ private void collectStats(Collection<String> labelNames, Collection<String> relT
                 }
             }
             try (ResourceIterator<Node> nodes = db.findNodes(label)) {
-                int count = 0;
-                while (nodes.hasNext() && count++ < SAMPLE) {
+                int count = 1;
+                while (nodes.hasNext()) {
                     Node node = nodes.next();
+                    if(count++ % config.getSample() == 0) {
```

Comment:
I don't think this is correct yet. As each node label has different counts we have to normalize the sample rate for each label. 

E.g. if you want 1000 samples
and you have 

- 100 Node A you want all
- 1000 Node B you want all
- 10000 Node C you want each 10th
- 1M Node D you want each 1000th
etc.